runtime/parachains/src/inclusion.rs

```diff
@@ -255,20 +262,24 @@ impl<T: Trait> Module<T> {
 		let threshold = availability_threshold(validators.len());
 
 		let mut freed_cores = Vec::with_capacity(n_bits);
-		for (para_id, pending_availability) in assigned_paras_record.into_iter()
+		for (para_id, pending_availability, commitments) in assigned_paras_record.into_iter()
 			.filter_map(|x| x)
-			.filter_map(|(id, p)| p.map(|p| (id, p)))
+			.filter_map(|(id, p, c)| p.and_then(|p| c.map(|c| (id, p, c))))
 		{
 			if pending_availability.availability_votes.count_ones() >= threshold {
 				<PendingAvailability<T>>::remove(&para_id);
+				<PendingAvailabilityCommitments>::remove(&para_id);
+
 				Self::enact_candidate(
 					pending_availability.relay_parent_number,
-					pending_availability.receipt,
+					pending_availability.descriptor,
+					commitments,
 				);
 
 				freed_cores.push(pending_availability.core);
 			} else {
 				<PendingAvailability<T>>::insert(&para_id, &pending_availability);
+				<PendingAvailabilityCommitments>::insert(&para_id, &commitments);
```

Comment:
commitments don't need to be updated here; they haven't changed from the bitfields.